### PR TITLE
Make clippy happy

### PR DESCRIPTION
**Pull request by sandreae**
Make `cargo-clippy` happy everywhere. Add clippy and format check to CI.
## 📋 Checklist

- [x] Add tests that cover your changes
- [x] Add this PR to the _Unreleased_ section in `CHANGELOG.md`
- [x] Link this PR to any issues it closes
- [x] New files contain a SPDX license header

### CODE REVIEW

**Comment by adzialocha** [comment]
Sooooo happy!

File: p2panda-rs/src/lib.rs [hunk 1]
```diff
@@ -45,7 +45,7 @@
 #![warn(
     missing_copy_implementations,
     missing_debug_implementations,
-    missing_doc_code_examples,
+    rustdoc::missing_doc_code_examples,
```
Comment (adzialocha) [inline p2panda-rs/src/lib.rs:48]: This is trickier: https://github.com/p2panda/p2panda/issues/120

File: p2panda-rs/src/identity/key_pair.rs [hunk 2]
```diff
@@ -128,11 +128,19 @@ impl KeyPair {
...
+        public_key.verify(message, signature)?;
         Ok(())
     }
 }
 
+impl Default for KeyPair {
+    fn default() -> Self {
+        let mut csprng: OsRng = OsRng {};
+        let key_pair = Ed25519Keypair::generate(&mut csprng);
+        Self(key_pair)
```
Comment (adzialocha) [inline p2panda-rs/src/identity/key_pair.rs:140]: Could we just call KeyPair::new() here to avoid duplication?

File: p2panda-rs/src/lib.rs (hunk 1 above)
Comment (sandreae) [reply p2panda-rs/src/lib.rs:48]: Ah! I thought there was a complication there, but it worked in the library, so i figured it was ok... I reverted it now :+1:

File: p2panda-rs/src/identity/key_pair.rs (hunk 2 above)
Comment (sandreae) [reply p2panda-rs/src/identity/key_pair.rs]: :+1:

**Pull request merged**